extensions/amp-experiment/1.0/amp-experiment.js

```diff
@@ -48,21 +50,43 @@ export class AmpExperiment extends AMP.BaseElement {
       const variantsService = responses[0];
       const enabled = responses[1];
 
-      if (!enabled) {
-        user().error(TAG, 'Experiment amp-experiment-1.0 is not enabled.');
+      let config = dict({});
 
-        // Ensure downstream consumers don't wait for the promise forever.
-        variantsService.init(Promise.resolve({}));
+      try {
+        config = this.getConfig_();
 
-        return Promise.reject('Experiment amp-experiment-1.0 is not enabled.');
-      }
+        if (!enabled) {
+          user().error(TAG, 'Experiment amp-experiment-1.0 is not enabled.');
+
+          // Ensure downstream consumers don't wait for the promise forever.
+          variantsService.init(
+            Promise.resolve(this.getEmptyExperimentToVariant_(config))
```

Comment:
question: Can we resolve with `Promise.resolve({})` instead. What is the difference between `{}` and `this.getEmptyExperimentToVariant_(config)`?

Reply:
So the reason why I changed this to `this.getEmptyExperimentToVariant_(config)`, is because doing `Promise.resolve({})` ends up throwing an error when trying to send out analytics ping.

Thus, I figured people would want to test that the ping is being sent out even when experiments are disabled. And since they are being disabled with `this.getEmptyExperimentToVariant_(config)`, it sets the param to equal `none`. Meaning the experiment was not applied 😄 

Let me know if this makes sense. I personally think that getting a ping that essentially says "all experiments are not being applied" is better than just throwing an error from the analytics side of things.

Reply:
> Let me know if this makes sense. I personally think that getting a ping that essentially says "all experiments are not being applied" is better than just throwing an error from the analytics side of things.

Oh definitely! A few thoughts: 
1. Do you think it's worth it to cache the emptyExperimentToVariant object so that we don't need to create it every time? Really depends on how many times do you expect the function to be called. 
2. In the case where 'amp-experiment-1.0' is not enabled. Do we expect that as user error? If so on the analytics side should we throw error or still send ping says "all experiments are not being applied"?

Reply:
@zhouyx 

> 1. Do you think it's worth it to cache the emptyExperimentToVariant object so that we don't need to create it every time? Really depends on how many times do you expect the function to be called.

Nope, because it is and always should only be called once. 😄 

> 2. In the case where 'amp-experiment-1.0' is not enabled. Do we expect that as user error? If so on the analytics side should we throw error or still send ping says "all experiments are not being applied"?

Ummmm I think we should still only throw an error on the experiment side. As this is an amp-experiment error, and not an amp-analytics error. And a ping saying all experiments are not being applied is not wrong. 😄